implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -138,6 +140,22 @@ void setExpectedAudience(JwtConsumerBuilder builder, JWTAuthContextInfo authCont
         }
     }
 
+    private void setRequiredClaims(JwtConsumerBuilder builder, JWTAuthContextInfo authContextInfo) {
```

Comment:
@radcortez We've discussed with @darranl before, and with Stuart in Quarkus, but in general we just prefer a simple old Java code which performs better. 
But in this case IMHO we don't need registering the custom verificators at all.
[Over here](https://github.com/smallrye/smallrye-jwt/blob/master/implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java#L93), you can add a simple function passing the parsed claims and context, `JwtClaims` has a `getClaimsAsMap` or similar, then the following one liner will do
```
if (!claims.getAsMap().keySet().containsAll(requiredClaims)) {
    throw new InvalidJwtException("Required claims are not available");
}
``` 

Something like that :-)